### PR TITLE
Replace distutils.version with packaging.version's parse

### LINKED ISSUE #35: Replace distutils.version with packaging.version.path
Using the pytest plugin on 3.11 (but 3.10 too) gives the following warning:

```
/home/runner/work/urllib3/urllib3/.nox/test-3-11/lib/python3.11/site-packages/pytest_freezegun.py:5: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  from distutils.version import LooseVersion
```

One option to fix this is to use packaging.version.path as done by @hugovk here: https://github.com/python-pillow/Pillow/pull/4797/files

Initial reported in  https://github.com/spulec/freezegun/issues/428 by mistake

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://github.com/ktosiek/pytest-freezegun/issues/35.

# Demo

```pycon
Python 3.10.0 (v3.10.0:b494f5935c, Oct  4 2021, 14:59:20) [Clang 12.0.5 (clang-1205.0.22.11)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from packaging.version import parse as parse_version
>>> import pytest
>>> parse_version(pytest.__version__)
<Version('6.2.5')>
>>> parse_version('3.6.0')
<Version('3.6.0')>
>>> parse_version(pytest.__version__) < parse_version('3.6.0')
False
```

### CODE REVIEW

**Review by pquentin — APPROVED** [review]
Thanks!

**Review by thijskramer — APPROVED** [review]

**Comment by eraserhead0705** [comment]
Any idea when we can see this released? Work around for me is currently manually changing the code in my venv.

**Comment by taybin** [comment]
I'm seeing this too now.  Any updates?

**Comment by lyz-code** [comment]
@ktosiek @thijskramer @pquentin is there anything we can do to speed up the merging of this PR?

**Comment by gabriel-yahav** [comment]
> @ktosiek @thijskramer @pquentin is there anything we can do to speed up the merging of this PR?

Joining the question. 
1. Is there any issue block merging this PR ?
2. When will a new release's version be available ?

**Comment by mgorny** [comment]
Honestly, at this point I do wonder if there's really a point in supporting pytest < 3.6.0, and if it wouldn't be better to just drop that and avoid the extra dep.

**Comment by hugovk** [comment]
Good idea, please see https://github.com/ktosiek/pytest-freezegun/pull/38 as an alternative.

**Comment by hugovk** [comment]
Closing this year old PR, happy to re-open if there's any interest, but will keep https://github.com/ktosiek/pytest-freezegun/pull/38 open a bit longer.

**Comment by youknowone** [comment]
@ktosiek would you try to recruit more maintainers?

**Pull request closed without merging**
